SR2019 floorplans

WillB97 opened the pull request:
Initial pull request of Student Robotics 2018 competition floorplans for setup and public use.
Contains floorplan templates and an SVG generation script to produce all required diagrams through hiding layers and merging template images.
Opened to allow collaboration on designs, see srobo/tasks#64 and  srobo/tasks#65.

WillB97 committed:
Added initial floorplan designs for the 2019 competition.

Included is a 1:1 representation of the areas of SUSU that have been booked.
- Dimensions still to verify are highlighted in red on venue related layers
- Initial area locations are marked without dimensions
- Power source and network locations are incomplete for the cafe

WillB97 committed:
Added MacOS gitignore

- Added .DS_Store to gitignore to prevent Mac Finder related status files

WillB97 committed:
Added key.svg and updated level 3 template

WillB97 committed:
Updated YAML layer spec

Converted 'embed' element to a list of dicts to allow for multiple images to be embedded

WillB97 committed:
Added svg generation script

Specification file can follow either the JSON or YAML format

WillB97 committed:
Added preliminary layout specification files

Also added exemplar team_names.txt

WillB97 committed:
Fix readme formatting

PeterJCLaw commented:
@WillB97 minor meta point: this year's competition is the "SR2019" competition -- our naming scheme is to use the calendar year in which the competition event happens.

PeterJCLaw commented on 2018/README.md:
```diff
@@ -0,0 +1,111 @@
+Locations highlighted in red on one of the venue related layers require measurements to be taken to assure their size.
+
+#### Requires:
+- \> python 3.4
```
The `generate_svg.py` script uses f-strings, which are in Python 3.6+ only. Should the f-strings be removed, or is this requirement wrong?

WillB97 replied:
This is an error, it should be 3.6

PeterJCLaw commented on 2018/README.md:
```diff
@@ -0,0 +1,111 @@
+  Lv2.svg
+  Cube.svg
+  key.svg
+  layout.svg
```
`layout.svg` seems to be missing and the other names don't seem to match the names of the files in the `templates` directory. Is this expected?
This seems to be an issue as the `venue_map.yaml` file expects the `layout.svg` template to exist.

(meta: I'd be tempted to drop this sort of detail from the REAMDE, including specific file names here is a recipe for them getting out of date).

PeterJCLaw commented:
I'm getting a syntax error trying to run this:
```
$ python3.6 ./generate_svg.py l2_public.yaml
Using yaml
Traceback (most recent call last):
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 263, in iterfind
    selector = _cache[cache_key]
KeyError: ('.//svg:text[svg:tspan="{{title}}" ]', (('', 'http://www.w3.org/2000/svg'), ('cc', 'http://creativecommons.org/ns#'), ('dc', 'http://purl.org/dc/elements/1.1/'), ('inkscape', 'http://www.inkscape.org/namespaces/inkscape'), ('rdf', 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'), ('sodipodi', 'http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd'), ('svg', 'http://www.w3.org/2000/svg')))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./generate_svg.py", line 60, in <module>
    root.find('.//svg:text[svg:tspan="{{title}}" ]', ns)[0].text = spec.get('title',spec['image'])
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 298, in find
    return next(iterfind(elem, path, namespaces), None)
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 277, in iterfind
    selector.append(ops[token[0]](next, token))
  File "/usr/lib/python3.6/xml/etree/ElementPath.py", line 233, in prepare_predicate
    raise SyntaxError("invalid predicate")
SyntaxError: invalid predicate
```

WillB97 committed:
Fixed svg embedding function

Changed markers in layout specifications to only use allowed characters
Added markers to templates for embedding key

WillB97 committed:
Reduce wall thickness to 5cm

WillB97 committed:
Added layout.svg

Needed to produce venue wide diagrams

WillB97 committed:
Added a scaling parameter

Allows for 1:x scaling on the overall image so produced images are appropriately sized.

WillB97 committed:
Added improvements discussed at hackday

- Moved battery charging to Cube stage
- Added staging area in Cube
- Added the volunteer area
- Moved reception to other side of hallway
- Added 34 individual pits, including on level 4 (these are 2.5x2.5m)

WillB97 commented:
I have been unable to reproduce your issue, if it is still occuring could you print the contents of `ns` on the preceding line.

WillB97 committed:
Added pit labels and required power locations

WillB97 committed:
Added displays

WillB97 committed:
Added network

WillB97 committed:
Increased power sockets in the cube and level 2

- Also relocated display in cafe to be more central
- Added tables to volunteer area

WillB97 committed:
Added photo studio layout and increased the size of reception

WillB97 committed:
Added global diagram labels

WillB97 committed:
Added zone dimensions

WillB97 committed:
Fixed team names being applied

RealOrangeOne commented:
Overall, this looks great! Commenting on an SVG like this is quite difficult, so i'm going to have to do it here (tick off once complete):
- [x] https://github.com/srobo/comp-floorplans/pull/3#issuecomment-468920724
- [ ] Can we get a barrier of some sort down the middle of the reception desks, which will only be used during check-in. It would be useful to ferry people towards 1 of the desks. Alternatively, if possible, an additional set of tables so we can have 4 rows of people checking in.
- [x] Files are stored in `2018
- [ ] Can we get a barrier between bar 3 and the TV studio? Would be useful so it's obvious competitors etc shouldn't go there.
- [x] I notice there's very little open are teams can use to test their robots in outside the arena. Newbury used the whole of floor 1. Any suggestions where teams might be able to use? Theoretically the actual arena and their pits are enough, but they're going to ask, so we might as well have an answer prepared!
- [ ] Is it worth putting a barrier on the stage to make sure people don't walk into / tamper with the battery charging?
- [ ] Charging all those batteries off just 2 wall sockets sounds like it could be _interesting_, we may need to pull the power from the other sockets on the stage.
- [ ] If we have scope for it, having a display in the volunteer area could be quite nice.

WillB97 committed:
Corrected 2018 to 2019

trickeydan commented:
>     * [ ]  Charging all those batteries off just 2 wall sockets sounds like it could be _interesting_, we may need to pull the power from the other sockets on the stage.

I hear rumours from @Tyler-Ward of a 32A socket nearby that location.

WillB97 commented:
> * [x]  [#3 (comment)](https://github.com/srobo/comp-floorplans/pull/3#issuecomment-468920724)
> * [x]  Files are stored in `2018

My bad, I thought I'd already corrected that

> * [ ]  I notice there's very little open area teams can use to test their robots in outside the arena. Newbury used the whole of floor 1. Any suggestions where teams might be able to use? Theoretically the actual arena and their pits are enough, but they're going to ask, so we might as well have an answer prepared!

The seating in Bar 3 can be moved so we could move a few bays to make an area for testing.

> * [ ]  If we have scope for it, having a display in the volunteer area could be quite nice.

Assuming there is a network socket in the room this shouldn't be to difficult, however SUSU only have 3 screens on wheels so we will have to decide where they are best used.

> I hear rumours from @Tyler-Ward of a 32A socket nearby that location.

There is a 32A socket in the corner of the cube so we need to make sure there is sufficient distro to break that down into multiple 13A sockets

WillB97 committed:
Added various barriers

WillB97 commented:
> * [x]  Can we get a barrier of some sort down the middle of the reception desks, which will only be used during check-in. It would be useful to ferry people towards 1 of the desks. Alternatively, if possible, an additional set of tables so we can have 4 rows of people checking in.
> * [x]  Can we get a barrier between bar 3 and the TV studio? Would be useful so it's obvious competitors etc shouldn't go there.
> * [x]  Is it worth putting a barrier on the stage to make sure people don't walk into / tamper with the battery charging?

I believe SUSU has rope barriers although this is worth checking in https://github.com/srobo/tasks/issues/67

Tyler-Ward commented:
With the charging power, according to my maths even if both the charger and charger PSU are only 50% efficient we can still theoretically run 32 chargers of a 13A socket (not that that is a good idea) but the two next to charging should be fine if we do 2 4ways with 4 4ways in each.

For score screens in Bar3 and the volunteer area we can use computer displays rather than large TVs if we are short that way we save the good screens for the main areas.

some extra minor fixes.
- [ ] add a large display screen inside the cube.
- [ ] add score display on bar 3 seating area

WillB97 commented:
> With the charging power, according to my maths even if both the charger and charger PSU are only 50% efficient we can still theoretically run 32 chargers of a 13A socket (not that that is a good idea) but the two next to charging should be fine if we do 2 4ways with 4 4ways in each.

Just because you can doesn't mean you should.
 
> * [ ]  add a large display screen inside the cube.

Where abouts in the cube? Beside staging or up on the stage?

Tyler-Ward commented:
The screen in staging is a small screen for shepherding and staging uses, the big one will want to go by the doors and this shows the same score display as the ones in the team areas

WillB97 committed:
Added additional screens

WillB97 committed:
Added cafe power sockets

WillB97 committed:
Finalised dimensions

WillB97 committed:
Added version to layouts and included produced output

Tyler-Ward reviewed (approved):
These are looking good to me. lets get these merged in.

mildlyincompetent reviewed (approved):
Looks good, we may want to make minor changes later but this is a good base.

PeterJCLaw commented:
@WillB97 thanks for preparing these. (I've finally found time and managed to work out how to view them)

Would it be possible to change the colour of the squares used to indicate displays? It's somewhat confusing that they're the same colour as fire exits.

PeterJCLaw commented:
> I have been unable to reproduce your issue, if it is still occuring could you print the contents of `ns` on the preceding line.

```
$ git diff
diff --git a/2019/generate_svg.py b/2019/generate_svg.py
index 54176ef..89d3035 100755
--- a/2019/generate_svg.py
+++ b/2019/generate_svg.py
@@ -56,6 +56,8 @@ root_tree = ET.parse('templates/' + spec['image']) # load svg
 root = root_tree.getroot()
 
 
+from pprint import pprint
+pprint(ns)
 try: # set title and version
   root.find('.//svg:text[svg:tspan="{{title}}"]', ns)[0].text = spec.get('title',spec['image'])
   root.find('.//svg:text[svg:tspan="{{version}}"]', ns)[0].text = "Version: " + str(spec.get('version',0.1))
```
yields
```
{'': 'http://www.w3.org/2000/svg',
 'cc': 'http://creativecommons.org/ns#',
 'dc': 'http://purl.org/dc/elements/1.1/',
 'inkscape': 'http://www.inkscape.org/namespaces/inkscape',
 'osb': 'http://www.openswatchbook.org/uri/2009/osb',
 'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#',
 'sodipodi': 'http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd',
 'svg': 'http://www.w3.org/2000/svg'}
```
